packages/local_auth/local_auth/lib/src/local_auth.dart

```diff
@@ -0,0 +1,182 @@
+  /// dialog after the face is recognized to make sure the user meant to unlock
+  /// their phone.
+  ///
+  /// Setting [biometricOnly] to true prevents authenticates from using non-biometric
+  /// local authentication such as pin, passcode, and passcode.
+  ///
+  /// Throws an [PlatformException] if there were technical problems with local
+  /// authentication (e.g. lack of relevant hardware). This might throw
+  /// [PlatformException] with error code [otherOperatingSystem] on the iOS
+  /// simulator.
+  @Deprecated('Use `requestAuthentication` instead.')
+  Future<bool> authenticate({
+    required String localizedReason,
+    bool useErrorDialogs = true,
+    bool stickyAuth = false,
+    AndroidAuthMessages androidAuthStrings = const AndroidAuthMessages(),
+    IOSAuthMessages iOSAuthStrings = const IOSAuthMessages(),
+    bool sensitiveTransaction = true,
+    bool biometricOnly = false,
+  }) {
+    return requestAuthentication(
+      localizedReason: localizedReason,
+      authMessages: <AuthMessages>[iOSAuthStrings, androidAuthStrings],
+      options: AuthenticationOptions(
+        useErrorDialogs: useErrorDialogs,
+        stickyAuth: stickyAuth,
+        sensitiveTransaction: sensitiveTransaction,
+        biometricOnly: biometricOnly,
+      ),
+    );
+  }
+
+  /// Authenticates the user with biometrics available on the device while also
+  /// allowing the user to use device authentication - pin, pattern, passcode.
+  ///
+  /// Returns true if the user successfully authenticated, false otherwise.
+  ///
+  /// [localizedReason] is the message to show to user while prompting them
+  /// for authentication. This is typically along the lines of: 'Please scan
+  /// your finger to access MyApp.'. This must not be empty.
+  ///
+  /// Provide [authMessages] if you want to
+  /// customize messages in the dialogs.
+  ///
+  /// Provide [options] for configuring further authentication related options.
+  ///
+  /// Throws a [PlatformException] if there were technical problems with local
+  /// authentication (e.g. lack of relevant hardware). This might throw
+  /// [PlatformException] with error code [otherOperatingSystem] on the iOS
+  /// simulator.
+  Future<bool> requestAuthentication(
+      {required String localizedReason,
+      Iterable<AuthMessages> authMessages = const <AuthMessages>[
+        IOSAuthMessages(),
+        AndroidAuthMessages()
+      ],
+      AuthenticationOptions options = const AuthenticationOptions()}) {
+    return LocalAuthPlatform.instance.authenticate(
+      localizedReason: localizedReason,
+      authMessages: authMessages,
+      options: options,
+    );
+  }
+
+  /// Returns true if auth was cancelled successfully.
+  /// This api only works for Android.
+  /// Returns false if there was some error or no auth in progress.
+  ///
+  /// Returns [Future] bool true or false:
+  Future<bool> stopAuthentication() async {
+    if (_platform.isAndroid) {
```

Comment:
Oops, this needs to be fixed; I missed it before. We should never do this kind of platform gating in the app-facing package.

We should instead just always pass through, and document that it is only supported on some platforms. The fact that it used to return `true` on iOS is weird since it's the opposite of what the docs say it would do. We should:
- Do a precursor PR to update `local_auth_ios` to return false, not throw (with a comment saying it's not supported), then:
- Remove this check.
- Make the minimum version of the iOS implementation required here the version with that fix.
- Explicitly list lack of platform support as one of the `false` cases in the comment above (arguably it's covered by "some error", but let's be explicit).

Reply:
Precursor PR: #5249